### DISCUSSION TITLE
How to access IOptions on adding HealthChecks in Startup?

### DISCUSSION THREAD

**Discussion by agentschmitt** [dotnet/aspnetcore#24477] [General]
How can i add HealthChecks with options from my IOptions that is registered in the servicecollection before?
In dotnet 2.1 it was possible to inject the options in Startup but with #9337 this was removed for dotnet 3.0
I saw many examples for HealthCheck registration using IConfiguration directly, but this is no good option for me, cause i have some placeholder replacement logic running for my IOptions. So is there any way to add healthchecks with configuration from IOptions?

```
public class Startup
    {
        private IOptions<MyApiOptions> ApiOptions { get; }

        public Startup(IOptions<MyApiOptions> apiOptions)
        {
            ApiOptions = apiOptions;
        }

        public void ConfigureServices(IServiceCollection services)
        {
            services
                .AddHealthChecks()
                .AddUrlGroup(new Uri(ApiOptions.Value.Url));

```

**Comment by davidfowl** [comment]
You can't access options that early. Seems like you need to delay the creation of the UrlGroup until they are ready.

**Comment by Tratcher** [comment]
I can't find the AddUrlGroup method, but most health check APIs eventually end up calling this:
https://github.com/dotnet/aspnetcore/blob/5a0526dfd991419d5bce0d8ea525b50df2e37b04/src/HealthChecks/HealthChecks/src/DependencyInjection/HealthChecksBuilder.cs#L25-L28

There are variants of this that would let you resolve services to configure your options.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1#use-di-services-to-configure-options

**Reply by davidfowl** [reply]
I think this is an old version of the healthchecks API

**Comment by agentschmitt** [comment]
Yeah thx i could use the optionspattern to add some healthchecks with usage
of ioptions.
Only downside is i cant use the healthchecksbuilder which is the target for most extension methods for registration.
And the addurlgroup is from https://github.com/xabaril/AspNetCore.Diagnostics.HealthChecks

David Fowler <notifications@github.com> schrieb am Mo., 3. Aug. 2020, 18:08:
